### ISSUE TITLE
Why don't we import paho mqtt c targets directly?

### ISSUE THREAD

**Issue by Aposhian** [eclipse/paho.mqtt.cpp#416]
Currently, the CMake finds paho.mqtt.c with a shim [`FindPahoMqttC.cmake`](https://github.com/eclipse/paho.mqtt.cpp/blob/master/cmake/FindPahoMqttC.cmake) file. This calls `find_library` on the desired library name, and [manually calls `find_path` on `MQTTAsync.h`](https://github.com/eclipse/paho.mqtt.cpp/blob/2ff3d155dcd10564f1816675789284b4efd79eb7/cmake/FindPahoMqttC.cmake#L18).

Why do we go through this effort when paho.mqtt.c already exports a config file (`eclipse-paho-mqtt-cConfig.cmake`) that exports its targets?

Basically my thought is that we should change `find_package(PahoMqttC)` to `find_package(eclipse-paho-mqtt-c)`.

If there is a technical/historical reason for this that I am missing, please let me know. I'm new to the project.

**Comment by fpagliughi** [comment]
Yes, sorry. The original CMake build was sent in on a PR, and the author moved on. At the time the C library's build was much different and didn't export the config.

I've been trying to get up to speed on CMake, with the intent of updating the builds for both project (C & C++) to make them both more modern and a little easier to use.

Alternately, I've been considering adding the whole C lib as a git submodule and building it together as a single library. I did something similar for the Paho Rust library (which also wraps the C lib), and it just made the whole thing much easier.

**Comment by fpagliughi** [comment]
Apologies, I had forgotten about this one, and in the meantime merged in an older PR #374 which added the Paho C submodule build. But the functionality was put into the [FindPahoMqttC.cmake](https://github.com/eclipse/paho.mqtt.cpp/blob/master/cmake/FindPahoMqttC.cmake) file!

So I will need to tear it apart to get this resolved. I will try to do so before the upcoming release.

**Comment by fpagliughi** [comment]
I put up a PR for the Paho C library to clean up the build there a little, and have a branch started to make use of those updates.
https://github.com/eclipse/paho.mqtt.c/pull/1428

If/when that lands, I'll move toward a v1.4.0 release to get the build cleaned up.

**Comment by fpagliughi** [comment]
I got tired of waiting for the upstream release, and worked around it for now. I'll probably rework it again after the next C update, but either way, this is basically implemented.
